scripts/data_overlap/compute_data_overlap_metrics.py

```diff
@@ -90,29 +90,65 @@ def create_ngram_index(
                 id = instance.id
                 assert id
                 input_tokens = tokenizer.tokenize(instance.input)
-                for input_ngram in ngrams(input_tokens, n):
-                    if input_ngram not in ngram_index[n]:
-                        ngram_index[n][input_ngram] = set()
-                    ngram_index[n][input_ngram].add(
+                for input_hash_info in get_ngram_hashes(input_tokens, n):
+                    input_hash, start, end = input_hash_info
+                    if input_hash not in ngram_index[n]:
+                        ngram_index[n][input_hash] = set()
```

Comment:
You can easily retrieve the original `input_ngram` here if you want to - it is simply the slice of `input_tokens` between the indexes `start` and `end`. Likewise below for `reference_unigrams`.